Code-the-Dream-School/cc-prac-team4-front
# PP-91-Pets-page-successfully-connected-with-back-end

This pull request adds functionality to retrieve all pets from the API and display them on the Pets page [http://localhost:3000/pets](http://localhost:3000/pets). It includes a new file petAction.js that contains the getPet function, which makes an API request to fetch all pets. The response data is then dispatched to update the state in the petsReducer file.

Additionally, the PetCard component has been modified to accept a pet prop and display the pet's information, such as name, gender, age, color, and breed. The component also dynamically loads different images based on the pet's type (cat or dog).

**Testing Instructions:**

- Install the necessary dependencies by running npm install in the project directory.
- Verify that the Pets page [http://localhost:3000/pets](http://localhost:3000/pets) displays a loader while the pets are being fetched.
- Once the pets are loaded, ensure that the pet cards are displayed with the correct information (name, gender, age, color, and breed).
- Check that the pet images are loaded correctly based on the pet's type (cat or dog).

<img width="1011" alt="Pets Page" src="https://github.com/AlonaVladymyrovaTrinity/adopt-a-pet/blob/main/PetsPage.png?raw=true">
